.github/actions/deploy-cloud-run/action.yml

```
@@ -0,0 +1,81 @@
+name: Deploy Cloud Run service
+description: Create a new revision and rollback if health check fails
+inputs:
+  region:
+    description: Region of the Cloud Run service
+    required: true
+  image:
+    description: Image name
+    required: true
+  service:
+    description: Name of the Cloud Run service
+    required: true
+  health_check_url:
+    description: Health check URL. Multiple URLs can be separated by line breaks. If not provided, health check will be skipped.
+
+runs:
+  using: composite
+  steps:
+    - name: Get current revision
+      id: get_revision
+      shell: bash
+      run: |
+        REV=$( \
+          gcloud run services describe "${{ inputs.service }}" \
+            --region "${{ inputs.region }}" \
+            --format="value(status.latestReadyRevisionName)" \
+        )
+        echo "Current revision: $REV"
+        echo "revision=$REV" >> "$GITHUB_OUTPUT"
+
+    - name: Deploy
+      shell: bash
+      run: |
+        gcloud run deploy "${{ inputs.service }}" \
+          --image "${{ inputs.image }}" \
+          --region "${{ inputs.region }}" \
+          --platform managed \
+          --quiet
+
+    # ensure that the traffic is updated to the latest revision always
+    - name: Update traffic
+      shell: bash
+      run: |
+        gcloud run services update-traffic "${{ inputs.service }}" \
+          --to-latest \
+          --region "${{ inputs.region }}"
```

We can later direct all traffic to `green` tag if healthcheck is okay

```suggestion
    - id: 'direct_green'
       name: Direct all traffics to green if healthcheck is passing
       if: ${{ steps.health_check.outputs.error != "true" }} # I believe `true` in health_check output is not parsed as bool but string
       uses: 'google-github-actions/deploy-cloudrun@v2'
       with:
         service: ${{ inputs.service }}
         image: ${{ inputs.image }}
         region: ${{ inputs.region }}
         tag_traffic: 'green=100' # this mean the new release will be having 100% traffic directed to it
     - name: Remove green tag
        needs: direct_green
        shell: bash
        run: |
          gcloud run services update-traffic "${{ inputs.service }}" \
            --remove-tags green \
            --region "${{ inputs.region }}"
```